includes/admin/admin-actions.php

```diff
@@ -630,4 +630,182 @@ function give_get_user_roles() {
 	endforeach;
 
 	return $user_roles;
-}
\ No newline at end of file
+}
+
+
+/**
+ * Ajax handle for donor address.
+ *
+ * @since 2.0
+ *
+ * @return string
+ */
+function __give_ajax_donor_manage_addresses() {
+	// Bailout.
+	if (
+		empty( $_POST['form'] ) ||
+		empty( $_POST['donorID'] )
+	) {
+		wp_send_json_error( array( 'error' => 1 ) );
+	}
+
+	$post                  = give_clean( wp_parse_args( $_POST ) );
+	$donorID               = absint( $post['donorID'] );
+	$form_data             = give_clean( wp_parse_args( $post['form'] ) );
+	$is_multi_address_type = ( 'billing' === $form_data['address-id'] || false !== strpos( $form_data['address-id'], '_' ) );
+	$address_type          = false !== strpos( $form_data['address-id'], '_' ) ?
+		array_shift( explode( '_', $form_data['address-id'] ) ) :
+		$form_data['address-id'];
+	$address_id            = false !== strpos( $form_data['address-id'], '_' ) ?
+		array_pop( explode( '_', $form_data['address-id'] ) ) :
+		null;
+	$response_data         = array(
+		'action' => $form_data['address-action'],
+		'id'     => $form_data['address-id'],
+	);
+
+	// Security check.
+	if( ! wp_verify_nonce( $form_data['_wpnonce'], 'give-manage-donor-addresses' ) ) {
+		wp_send_json_error( array(
+				'error' => 1,
+				'error_msg' => wp_sprintf(
+					'<div class="notice notice-error"><p>%s</p></div>',
+					__( 'Error: Security issue. please try after sometime.', 'give' )
```

Comment:
Remove "Please try after some time" that doesn't make sense and only frustrates users. Trying after sometime won't change the end result.